--- a/qtdeclarative/tools/CMakeLists.txt
+++ b/qtdeclarative/tools/CMakeLists.txt
@@ -37,49 +37,6 @@
 if(QT_FEATURE_qml_preview AND QT_FEATURE_thread AND NOT ANDROID AND NOT WASM AND NOT IOS AND NOT rtems)
     add_subdirectory(qmlpreview)
 endif()
-if(QT_BUILD_SHARED_LIBS AND QT_FEATURE_thread AND TARGET Qt::Quick AND NOT ANDROID AND NOT WASM AND NOT IOS AND NOT rtems)
-    add_subdirectory(qmlscene)
-    add_subdirectory(qmltime)
-endif()
-if(QT_BUILD_SHARED_LIBS
-        AND QT_FEATURE_process
-        AND QT_FEATURE_regularexpression
-        AND QT_FEATURE_thread
-        AND TARGET Qt::Quick
-        AND NOT ANDROID
-        AND NOT IOS
-        AND NOT WASM
-        AND NOT rtems)
-    add_subdirectory(qmlplugindump)
-endif()
-if(TARGET Qt::Quick
-        AND TARGET Qt::Widgets
-        AND QT_FEATURE_checkbox
-        AND QT_FEATURE_combobox
-        AND QT_FEATURE_dialogbuttonbox
-        AND QT_FEATURE_formlayout
-        AND QT_FEATURE_groupbox
-        AND QT_FEATURE_lineedit
-        AND QT_FEATURE_mainwindow
-        AND QT_FEATURE_spinbox
-        AND QT_FEATURE_textedit
-        AND NOT ANDROID
-        AND NOT IOS
-        AND NOT WASM
-        AND NOT rtems)
-    add_subdirectory(qmleasing)
-endif()
-if(TARGET Qt::Quick
-        AND TARGET Qt::Svg
-        AND NOT ANDROID
-        AND NOT IOS
-        AND NOT WASM
-        AND NOT rtems)
-    add_subdirectory(svgtoqml)
-endif()
-if(QT_FEATURE_thread AND TARGET Qt::QuickTest AND NOT ANDROID AND NOT WASM AND NOT rtems)
-    add_subdirectory(qmltestrunner)
-endif()
 if(QT_FEATURE_private_tests AND QT_FEATURE_thread AND NOT ANDROID AND NOT WASM AND NOT IOS AND NOT rtems)
     add_subdirectory(qmljs)
 endif()
